content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,241 @@
+
+Some errors cannot be overcome by retries and require changes to the request.
+
+#### Queries
+Many of the `GET` requests have specific parameter requirements. The [SDKs](../software-and-sdks/) can help enforce them, but you can still pass invalid arguments (e.g. an asset string that isn't [SEP-11 conformant](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md#asset)) that error out every time. In this scenario, there's nothing you can do aside from following the [API specification](../../api) precisely. The `extras` field of the error response will often clue you in on where to look and what to look for:
+
+<CodeExample>
+
+```bash
+curl -s https://horizon-testnet.stellar.org/claimable_balances/0000 | jq '.extras'
+{
+  "invalid_field": "id",
+  "reason": "Invalid claimable balance ID"
+}
+```
+
+</CodeExample>
+
+Note that the SDKs make it a point to distinguish an **invalid request** (as above) vs. a **missing resource** (a `404 Not Found`) (e.g. the generic `NetworkError` vs. a `NotFoundError` in the JavaScript SDK), where the latter might not be considered an error depending on your situation.
+
+#### Safe Transaction Adjustments
+Certain transaction submission failures also need adjustments to succeed. If the XDR is malformed or the transaction is somehow otherwise invalid, you'll encounter a `400 Bad Request` (for example, when excluding a source account in the [API reference example](../../api/errors/response/)). Look at the `extras.result_codes` field for details and cross-reference them with the appropriate [Result Codes documentation](../../api/errors/result-codes/) to determine what went wrong.
+
+Another class of safe adjustments involves transaction fees: if you get a `tx_insufficient_fee` error, it's worth reading the [later section](#insufficient-fees-and-surge-pricing) to adjust your fee-paying strategy.
+
+#### Unsafe Transaction Adjustments
+As outlined in the section on [retries](#retrying-until-success), resubmitting an *unchanged* (and valid) transaction (with the same operations, signatures, sequence number, etc.) is always safe to do. You should be careful when working around an error that *does* require changes to the transaction, though: it's very possible to cause **duplication transactions** which can result in all sorts of problems (double-payments, erroneous trustlines, etc.).
+
+##### Example: Invalid Sequence Numbers
+These errors typically occur when you have an outdated view of an account due to other transactions happening outside of your worldview. This could be because the account is used on multiple devices, you have concurrent submissions happening, or a number of other reasons. Thankfully, the solution is **usually** relatively simple: retrieve the account details and try again with an updated sequence number.
+
+<CodeExample>
+
+```js
+// suppose `account` is an outdated `AccountResponse` object
+let tx = sdk.TransactionBuilder(account, ...)/* etc */.build()
```

Comment:
The `...` here isn't valid JS, which might break some random syntax highlighting bits. Not the worst thing in the world, but thought I'd point it out while looking!

Reply:
Looks like it renders okay, but thanks for noting it! :+1: